src/rules/groupedImportsRule.ts

```diff
@@ -0,0 +1,152 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Lint from "tslint";
+import { isImportDeclaration } from "tsutils";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "grouped-imports",
+        description: "Separate import groups by blank lines.",
+        rationale: "Keeps a clear overview on dependencies.",
+        optionsDescription: "Not configurable.",
+        hasFix: true,
+        options: {},
+        optionExamples: [true],
+        type: "style",
+        typescriptOnly: false,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static IMPORT_SOURCES_SEPARATED = "Import sources within a group must not be separated by blank lines";
+    public static IMPORT_SOURCES_NOT_SEPARATED =
+        "Import sources of different groups must be separated by a single blank line";
+    public static IMPORT_SOURCES_ORDER =
+        "Import sources of different groups must be sorted by: libraries, parent directories, current directory";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new Walker(sourceFile, this.ruleName, this.getOptions()));
+    }
+}
+
+enum ImportStatementType {
+    LIBRARY_IMPORT = 1,
+    PARENT_DIRECTORY_IMPORT = 2,
+    CURRENT_DIRECTORY_IMPORT = 3,
+}
+
+interface ImportStatement {
+    statement: ts.Statement;
+    type: ImportStatementType;
+    lineStart: number;
+    lineEnd: number;
+}
+
+class Walker extends Lint.AbstractWalker<Lint.IOptions> {
+    private lastImportStatement: ImportStatement;
+
+    private static getImportStatementType(statement: ts.Statement): ImportStatementType {
+        const path = Walker.getImportPath(statement);
+        if (path.charAt(0) === ".") {
+            if (path.charAt(1) === ".") {
+                return ImportStatementType.PARENT_DIRECTORY_IMPORT;
+            } else {
+                return ImportStatementType.CURRENT_DIRECTORY_IMPORT;
+            }
+        } else {
+            return ImportStatementType.LIBRARY_IMPORT;
+        }
+    }
+
+    private static getImportPath(statement: ts.Statement): string {
+        const str = statement.getText();
+        let index;
+        let lastIndex;
+        index = str.indexOf("'");
+        if (index > 0) {
+            lastIndex = str.lastIndexOf("'");
+        } else {
+            index = str.indexOf("\"");
+            lastIndex = str.lastIndexOf("\"");
+        }
+        if (index < 0 || lastIndex < 0) {
+            throw new Error(`Unable to extract path from import statement \`${statement.getText()}\``);
+        }
+        return str.substring(index + 1, lastIndex);
+    }
+
+    public walk(sourceFile: ts.SourceFile): void {
+        sourceFile.statements
+            .filter(isImportDeclaration)
```

Comment:
what about `ImportEqualsDeclaration` where `isExternalModuleReference(statement.moduleReference)`? For example:

```ts
import fs = require('fs');
```